### PR TITLE
Consistency prompting

### PR DESCRIPTION
**Pull request by thejaminator**

### CODE REVIEW

File: cot_transparency/data_models/models.py [hunk 1]
```diff
@@ -117,6 +118,7 @@ class TaskSpec(BaseModel):
     out_file_path: Path
     ground_truth: MultipleChoiceAnswer
     formatter_name: str
+    intervention_name: Optional[str] = None
```
Comment (thejaminator) [inline cot_transparency/data_models/models.py:121]: specify a few shot intervention!

File: stage_one.py [hunk 2]
```diff
@@ -66,15 +68,29 @@
...
+        for intervention in interventions:
+            with_interventions.append(
+                TaskSetting(
+                    task=setting.task,
+                    formatter=setting.formatter,
+                    model=setting.model,
+                    intervention=intervention,
+                )
+            )
+    return task_settings + with_interventions
```
Comment (thejaminator) [inline stage_one.py:93]: for each possible intervention, create a corresponding setting

File: stage_one.py [hunk 3]
```diff
@@ -121,37 +137,46 @@ def get_list_of_examples(task: str, dataset: Optional[str] = None) -> list[DataE
...
         model = setting.model
         formatter = setting.formatter
+        print(f"formatter: {formatter.name()}")
         data: list[DataExampleBase] = get_list_of_examples(task, dataset=dataset)
-        out_file_path: Path = Path(f"{exp_dir}/{task}/{model}/{formatter.name()}.json")
+        out_file_path: Path = (
+            Path(f"{exp_dir}/{task}/{model}/{formatter.name()}.json")
+            if setting.intervention is None
+            else Path(f"{exp_dir}/{task}/{model}/{setting.intervention.name()}_on_{formatter.name()}.json")
+        )
```
Comment (thejaminator) [inline stage_one.py:179]: therefore the name becomes like `BiasedConsistency10_on_ZeroShotCOTUnbiasedFormatter.json`

File: tests/test_answer_finding.py [hunk 4]
```diff
@@ -20,6 +22,23 @@ def test_extract_answer(test_case: dict[str, str]):
...
+
+Step 2: Calculate the date 24 hours later
+Adding 24 hours to May 30, 2021, gives us May 31, 2021.
+
+Step 3: Verify the date format
+The format requested is MM/DD/YYYY.
+Actually I made a mistake
+The best answer is: (B) 05/31/2021.
+"""
+    assert extract_answer(long) == "B"
```
Comment (thejaminator) [inline tests/test_answer_finding.py:39]: just an extra test to make sure its really taking the last answer

File: cot_transparency/formatters/interventions/few_shots_loads.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+# Data previously generated with cot-transparency/scripts/dump_correct_cot_data.py
+# small brain cache to load only when needed
+@lru_cache
+def get_correct_cots() -> Slist[TaskOutput]:
+    jsons_tasks: Slist[TaskOutput] = read_jsonl_file_into_basemodel(
+        Path("data/bbh_correct_cots/data.jsonl"), TaskOutput
+    )
+
+    only_correct_cots: Slist[TaskOutput] = jsons_tasks
+    return only_correct_cots
```
Comment (thejaminator) [inline cot_transparency/formatters/interventions/few_shots_loads.py:19]: ezpz lazy loading

File: scripts/dump_correct_cot_data.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        # only get the ones that are correct
+        .filter(lambda x: x.model_output.parsed_response == x.task_spec.ground_truth)
+        .distinct_by(lambda x: x.task_spec.task_hash)
+    )
+    score = jsons_tasks.map(lambda x: x.is_correct).average()
+    print(f"Average score: {score}")
+    assert score == 1.0, f"This should be 1.0, got {score}"
+
+    print(f"Number of jsons: {len(jsons_tasks)}")
+    write_jsonl_file_from_basemodel(path=Path("data/bbh_correct_cots/data.jsonl"), basemodels=jsons_tasks)
```
Comment (thejaminator) [inline scripts/dump_correct_cot_data.py:47]: we need to dump good COT data too. i'm just gonna put it in `data/bbh_correct_cots/data.jsonl` if thats ok

File: accuracy.csv [hunk 7]
```diff
@@ -0,0 +1,33 @@
+,task_name,model,formatter_name,is_correct,accuracy_standard_error
```
Comment (erees1) [inline accuracy.csv:1]: Did you mean to merge this?

File: cot_transparency/data_models/models.py [hunk 8]
```diff
@@ -136,6 +138,10 @@ class TaskOutput(BaseModel):
     task_spec: TaskSpec
     model_output: ModelOutput
 
+    @property
+    def is_correct(self) -> bool:
```
Comment (erees1) [inline cot_transparency/data_models/models.py:142]: 🚀

File: stage_one.py (hunk 3 above)
Comment (erees1) [reply stage_one.py:180]: Love how long the names are getting

**Review by erees1 — APPROVED** [review]

File: stage_one.py (hunk 3 above)
Comment (thejaminator) [reply stage_one.py:180]: sad

File: accuracy.csv (hunk 7 above)
Comment (thejaminator) [reply accuracy.csv:1]: shit

**Pull request merged**